### ISSUE TITLE
Is there a way to implement this into the rich presence app?

### ISSUE THREAD

**Issue by saieeshb** [parampaa2/genshingrab#3]
I was wondering if there was a way to feed the date from the OCR to the rich presence app.  Is it possible?

**Comment by maufirf** [comment]
I'm not sure what do you mean by date, but I assume is the date like "Jun 28th, 20211" or something like that.

tl;dr answer: Yes, as long as you know where to obtain the date information.

I'm not sure on the step of obtaining dates through OCR as there's almost no sources of it in the Genshin Impact HUD or menu GUIs. However, if there's any, it is completely possible. The caveat is, the use of Tessaract OCR here is, for now, suited to only catch letters, especially for character names. Moreover, the GUI that had date displayed may only be accessed if the player decided to open that menu GUI that have the date displayed.

And for the RPC itself, as long as you can acquire the date, you can feed it to any other programs, and not just the RPC. Hell, you may even make a Facebook bot that posts everytime you died.

Despite all that, I don't really have a good idea to answer why would we put date on RPC, due to the fact that:

- There's more relevant information to be displayed on the RPC, like what AR the player is, in what area they are (Monstadt, Liyue, Inazuma, Dragonspine, etc.), or any others.
- People can just see their own date through their phone or calendar app, maybe?
- The number of RPC lines which we can put informations are limited.

**Comment by saieeshb** [comment]
Well my bad. I meant data(all the stuff you mentioned in the last part) but never realized the typo. 😅
